src/mgr/ActivePyModules.cc

```diff
@@ -1516,16 +1516,42 @@ void ActivePyModules::register_client(std::string_view name, std::string addrs)
   entity_addrvec_t addrv;
   addrv.parse(addrs.data());
 
-  dout(7) << "registering msgr client handle " << addrv << dendl;
-  py_module_registry.register_client(name, std::move(addrv));
+  dout(7) << "registering msgr client handle [" << name << "] " << addrv << dendl;
+  epoch_t e;
+  if (cluster_state.is_client_in_mgrmap(name, addrv, &e)) {
+    // Client already registered in the MgrMap
+      return;
+  }
+  py_module_registry.register_client(name, addrv);
+  while (true) {
+    // Wait to receive MgrMap from the monitor
+    C_SaferCond c;
+    cluster_state.wait_for_mgrmap(&c, e+1);
+    c.wait();
+    if (cluster_state.is_client_in_mgrmap(name, addrv, &e)) {
+      // Client registered in the updated MgrMap
+      return;
+    }
+    // While waiting to receive new MgrMap from the monitor, the
+    // client may be deregistered from the py_module_registry. If this
```

Comment:
The hint that it's expected to be done by another module is also useful as an explanation for why this extra check is not needed in a similar loop in `src/mgr/Mgr.cc` -- because sqlite just got initialized and no modules can be running at that point.  At least that is my assumption for why it's not needed there?

Reply:
> why this extra check is not needed in a similar loop in `src/mgr/Mgr.cc` -- because sqlite just got initialized and no modules can be running at that point

@batrick Can you confirm this understanding is correct?

Reply:
Yes, that's right.